docs/concepts/storage.md

```diff
@@ -0,0 +1,97 @@
+---
+id: storage
+title: Storage Staking
+sidebar_label: Storage Staking
+---
+
+If you’re familiar with Ethereum’s pricing model, you may know that the protocol charges a fee (called "gas") once for a transaction, and this transaction takes into account the amount of data stored via that transaction. This essentially means that anyone can pay once to store permanent data on-chain. This is a poor economic design for at least two reasons: 1. The people running the network (miners, in the case of Ethereum 1) are not appropriately incentivized to store large amounts of data, since a gas fee far charged in the distant past can increase storage costs forever, and 2. The users of a smart contract are charged for the data they send to store in it, rather than charging the owner of the smart contract.
+
+NEAR addresses this economic misalignment using a mechanism called storage staking (you may have heard that NEAR charges rent for storage – this has not been true [since March 2020](https://github.com/nearprotocol/nearcore/pull/2272)). In storage staking (sometimes called _state_ staking), the account that owns a smart contract must stake (or lock) tokens according to the amount of data stored in that smart contract, effectively reducing the balance of the contract's account.
+
+## How does this align incentives?
+
+Storage-staked tokens are unavailable for other uses such as validation staking. This increases the yield that validators will receive. Learn more in [the economics whitepaper](https://near.org/papers/economics-in-sharded-blockchain/).
+
+## When do tokens get staked?
+
+On each incoming transaction that adds data.
+
+Let's walk through an example:
+
+1. You launch a [a custom token](https://examples.near.org/?q=fungible), deploying your token's smart contract to the account `mycoiniscoolerthanyours.near`
+2. Token holders can name an escrow – an account which can transfer tokens on their behalf. Naming an escrow will require a call to your smart contract, and your smart contract will need to store this mapping of token-owner-to-escrow-account.
+3. When this call comes in, NEAR will check that `mycoiniscoolerthanyours.near` has a large enough balance that it can stake an amount corresponding to this data addition. If it does not, the transaction will fail. Otherwise, these new tokens get locked for storage staking and the transaction completes.
```

Comment:
can we use a different contract as an example? It can be confusing which tokens you are staking (NEAR vs mycoiniscoolerthanyours) Even switching to NFT will make it easier to grok.

Reply:
This cleaned it up a lot; great suggestion